sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/AADIssuerJWSKeySelector.java

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.spring.aad;
+
+
+import com.azure.spring.autoconfigure.aad.AADTokenClaim;
+import com.nimbusds.jose.JWSHeader;
+import com.nimbusds.jose.KeySourceException;
+import com.nimbusds.jose.jwk.source.JWKSource;
+import com.nimbusds.jose.jwk.source.RemoteJWKSet;
+import com.nimbusds.jose.proc.JWSAlgorithmFamilyJWSKeySelector;
+import com.nimbusds.jose.proc.JWSKeySelector;
+import com.nimbusds.jose.proc.SecurityContext;
+import com.nimbusds.jose.util.DefaultResourceRetriever;
+import com.nimbusds.jwt.JWTClaimsSet;
+import com.nimbusds.jwt.proc.JWTClaimsSetAwareJWSKeySelector;
+import java.net.URL;
+import java.security.Key;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Selecting key candidates for processing a signed JWT which provides access to the JWT claims set in addition to the
+ * JWS header.
+ */
+public class AADIssuerJWSKeySelector implements JWTClaimsSetAwareJWSKeySelector<SecurityContext> {
+
+    private AADTrustedIssuerRepository trustedIssuerRepository;
+
+    private int connectTimeout;
+
+    private int readTimeout;
+
+    private int sizeLimit;
+
+    private final Map<String, JWSKeySelector<SecurityContext>> selectors = new ConcurrentHashMap<>();
+
+    public AADIssuerJWSKeySelector(AADTrustedIssuerRepository trustedIssuerRepository, int connectTimeout,
+        int readTimeout, int sizeLimit) {
+        this.trustedIssuerRepository = trustedIssuerRepository;
```

Comment:
We could use `trustedIssuers` as the field instead of a `trustedIssuerRepository`. 

```java
private Set<String> trustedIssuers; 
```